File: ruby/bob/bob.rb

class Bob
  def hey(text)
    if say_nothing(text)
      'Fine. Be that way!'
    elsif yell(text)
      'Whoa, chill out!'
    elsif question(text)
      'Sure.'
    else
      'Whatever.'
    end
  end

  private

  def question(text)
    text[-1] == '?'
  end

  def yell(text)
    text.upcase == text && text.match(/[a-zA-Z]/)
  end

  def say_nothing(text)
    text.strip.empty?
  end
end

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {

  if (!input.match(/\S/)) {
    return "Fine. Be that way!";
  } else if (input.toUpperCase() == input && input.match(/[A-Z]/)) {
    return 'Whoa, chill out!';
  } else if (input.endsWith("?")) {
    return 'Sure.';
  } else {
    return 'Whatever.';
  }
};

module.exports = Bob;
